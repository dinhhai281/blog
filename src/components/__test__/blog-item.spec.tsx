import React from 'react';
import BlogItem, { BlogItemProps } from '@components/blog-item';
import { render } from '@utils/testing';

describe('BlogItem Component', () => {
  let props: BlogItemProps;

  beforeEach(() => {
    props = {
      source: {
        id: 'id',
        frontmatter: {
          date: '2020-07-12T00:00:00.000Z',
          title: 'title',
          path: '/path',
        },
        excerpt: 'excerpt',
      },
    };
  });

  it('render correctly', () => {
    const { asFragment } = render(<BlogItem source={props.source} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
