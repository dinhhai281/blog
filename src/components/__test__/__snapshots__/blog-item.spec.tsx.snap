// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogItem Component render correctly 1`] = `
<DocumentFragment>
  <div
    class="grid grid-cols-1"
  >
    <a
      class="text-pink-700 text-xl md:text-2xl font-semibold"
      data-testid="title"
      href="/en/path"
    >
      title
    </a>
    <div
      class="text-sm text-pink-700 text-opacity-75"
    >
      0amte_for0amt
    </div>
    <div
      class="pt-3"
    >
      excerpt
    </div>
  </div>
</DocumentFragment>
`;
